pyqtorch/analog.py

```diff
@@ -359,7 +361,9 @@ def __init__(
         self.qubit_support = operations.qubit_support
         self.n_qubits = n_qubits
 
-    def run(self, state: Tensor, values: dict[str, Tensor]) -> Tensor:
+    def forward(
+        self, state: Tensor, values: dict[str, Tensor] | ParameterDict = dict()
```

Comment:
its a bit strange to me that the forward of an  Add block does not add but just do the normal sequence forward

Reply:
Sorry, you are right. The point was actually that the `forward` of the observable should be the same as the `forward` from `Add`, so actually there should no be `forward` rewrite on the `Observable` because now it inherits from `Add`.

The `DiagonalObservable` currently inherits from `Primitive` so it should be handled differently.

Reply:
I see what you mean. Then it is better to remove the forward method to make use the one from Add.

Reply:
Now this should be handled @dominikandreasseitz @jpmoutinho.